### DISCUSSION TITLE
Local Modal Resolution without file extensions

### DISCUSSION THREAD

**Discussion by jonahallibone** [swc-project/swc#4722] [Ideas]
I know using bundlers (micro-bundle, babel, etc) it's possible to import local ESModules without needing to include file extensions. Specifically, im looking to replicate the import behavior I'm used to in React projects with TypeScript, but on the Node side.

Right now in my node app with TS is this
`import { addParams } from "./utils/params.js`

I'm looking to be able to have
`import { addParams } from "./utils/params`

Is this possible with the current configuration settings, or would this need to be a feature (and would it even be valid as I know the Node spec specifically calls for URL-like imports, and TS follows the lead on this).

Thanks!

**Comment by juanlatorre** [comment]
I'm currently trying to do the same thing, maybe a custom plugin to do something like this ? It would be great

`import x from "./folder"` -> `import x from "./folder/index.js"`
`import x from "./file"` -> `import x from "./file.js"`

Btw, I'm curring doing this as a workaround

<img width="766" alt="image" src="https://user-images.githubusercontent.com/4494526/171516898-c00f54d0-b246-4a58-8bf3-b5de13363c68.png">

**Comment by clayrisser** [comment]
I'm looking for the same thing
